src/interpreter.rs

```diff
@@ -76,7 +76,7 @@ impl Debug for Interpreter {
 }
 unsafe impl Send for Interpreter {}
 
-impl Interpreter {
+impl<'a> Interpreter {
```

Comment:
No need to add `'a` *here*. instead, maybe put it a generic argument on `input` function?

Reply:
You are right, thanks for the review. I will update it.

Reply:
Maybe I can remove it totally, because it compiler can assign those.